cfme/tests/control/test_actions.py

```diff
@@ -196,8 +204,16 @@ def automate_role_set(request):
 
 
 @pytest.fixture(scope="module")
-def vm_crud(vm_name, provider_crud):
-    return Vm(vm_name, provider_crud)
+def vm_crud(vm_name, provider_crud, provider_key, provider_type):
+    if is_cloud_provider(provider_key):
+        if provider_type == "openstack":
+            return OpenStackInstance(vm_name, provider_crud)
+        elif provider_type == "ec2":
+            return EC2Instance(vm_name, provider_crud)
```

Comment:
If that doesn't work out, I think I'd be alright if we just got all the cloud providers merged for now.